### PR TITLE
Add integration tests for PR#291, PR#292, PR#296

### PR DESCRIPTION
**Pull request by sivanov-nuodb**

### CODE REVIEW

**Review by bkelly-ndb — APPROVED** [review]

File: test/integration/template_admin_test.go [hunk 1]
```diff
@@ -371,6 +371,47 @@ func TestConfigDoesNotContainEmptyBlocks(t *testing.T) {
...
+
+	t.Run("testWithConfig", func(t *testing.T) {
+		options.SetValues = make(map[string]string)
+		options.SetValues["admin.configFiles.foo\\.conf"] = "foo"
+		// Run RenderTemplate to render the template and capture the output.
+		output := helm.RenderTemplate(t, options, helmChartPath, "release-name", []string{"templates/statefulset.yaml"})
+
+		for _, obj := range testlib.SplitAndRenderStatefulSet(t, output, 1) {
+			cksum = obj.Spec.Template.ObjectMeta.Annotations["checksum/config"]
+			assert.NotEmpty(t, cksum)
```
Comment (adriansuarez) [inline test/integration/template_admin_test.go:398]: Should we check that it is non-0?

File: test/integration/template_database_test.go [hunk 2]
```diff
@@ -1682,3 +1682,107 @@ func TestDatabaseIngressRenders(t *testing.T) {
...
+	})
+
+	t.Run("testWithConfig", func(t *testing.T) {
+		options.SetValues["database.configFiles.foo\\.conf"] = "foo"
+		// render the SMs and capture the output
+		output := helm.RenderTemplate(t, options, helmChartPath, "release-name", []string{"templates/statefulset.yaml"})
+
+		for _, obj := range testlib.SplitAndRenderStatefulSet(t, output, 1) {
+			cksum[obj.Name] = obj.Spec.Template.ObjectMeta.Annotations["checksum/config"]
+			assert.NotEmpty(t, cksum[obj.Name])
```
Comment (adriansuarez) [inline test/integration/template_database_test.go:1719]: Same comment here and on line 1727 about checking that the value is non-0.

**Review by adriansuarez — APPROVED** [review]
Looks good. I have a couple of minor comments about checking the hash value.

File: test/integration/template_admin_test.go (hunk 1 above)
Comment (sivanov-nuodb) [reply test/integration/template_admin_test.go:398]: Correct. I will add this check.

**Pull request merged**